### PR TITLE
publish artifact to bintray after successful build

### PR DESCRIPTION
**Pull request by hardproblems**
- Follow what aerosolve did to set up auto-deploy of tagged builds ( https://github.com/airbnb/aerosolve/commit/67280922b367aced4471894cd434ef540cff400b)

- Update the build section of README.md to include more complete instructions for manually uploading to jfrog

cc @alexism

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -76,8 +76,8 @@ allprojects {
   }
 
   bintray {
-    user = project.hasProperty('bintrayUser') ? project.getProperty('bintrayUser') : null
-    key  = project.hasProperty('bintrayKey')  ? project.getProperty('bintrayKey')  : null
+    user = System.env.BINTRAY_USER ?: bintray_user
```
Comment (alexism) [inline build.gradle]: where is `bintray_user` set?

Comment (hardproblems) [inline build.gradle]: ah they are set in gradle.properties in aerosolve...bad copy/pasting on my part

**Comment by hardproblems** [comment]
Updated build.gradle to remove the default value. TIL about this operator `?:` with a groovy name  http://docs.groovy-lang.org/latest/html/documentation/#_elvis_operator

File: build.gradle [hunk 2]
```diff
@@ -76,8 +76,8 @@ allprojects {
   }
 
   bintray {
-    user = project.hasProperty('bintrayUser') ? project.getProperty('bintrayUser') : null
-    key  = project.hasProperty('bintrayKey')  ? project.getProperty('bintrayKey')  : null
+    user = System.env.BINTRAY_USER ?: null
```
Comment (alexism) [inline build.gradle]: Being able to set it with a property is useful. Also I wonder how many know about the Elvis operator. Readability trumps conciseness imho.
I'd say:
`user = System.env.BINTRAY_USER ? System.env.BINTRAY_USER : project.getProperty('bintrayUser')`

**Comment by alexism** [comment]
LGTM

**Pull request merged**